refactored and optimized code

jeevesh2002 opened the pull request:
<!--
⚠️ If you do not respect this template, your pull request will be closed.
⚠️ Your pull request title should be short detailed and understandable for all.
⚠️ Also, please add a release note file using reno if the change needs to be
  documented in the release notes.
⚠️ If your pull request fixes an open issue, please link to the issue.

- [ ] I have added the tests to cover my changes.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the CONTRIBUTING document.
-->

### Summary
Refactored and optimized code according to the zen of python.


### Details and comments
line 58 - removed the unnecessary call to str() within print()
line 68 - replaced interpolated string formatting with f -string, 
               simplified unnecessary nesting, casting, and constant values in f -strings.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/Qiskit/qiskit-terra?pullRequest=9292) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/Qiskit/qiskit-terra?pullRequest=9292) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/Qiskit/qiskit-terra?pullRequest=9292) it.</sub>

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/Qiskit/qiskit-terra?pullRequest=9292) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/Qiskit/qiskit-terra?pullRequest=9292) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/Qiskit/qiskit-terra?pullRequest=9292) it.</sub>

jeevesh2002 commented:
### Summary

Refactored and optimized code according to the zen of python.
### Details and comments

line 58 - removed the unnecessary call to str() within print()
line 68 - replaced interpolated string formatting with f -string,
simplified unnecessary nesting, casting, and constant values in f -strings.
grover.py  merged else clause's nested if statement into elif

woodsp-ibm commented on qiskit/algorithms/amplitude_amplifiers/grover.py:
```diff
@@ -310,35 +310,34 @@ def amplify(self, amplification_problem: AmplificationProblem) -> "GroverResult"
+                top_measurement = np.binary_repr(max_amplitude_idx, num_bits)
+                max_probability = np.abs(max_amplitude) ** 2
+                shots = 1
             else:
```
The use of quantum_instance with grover is deprecated - this is if/else under the else branch depending on the the simulation type (statevector or not). This code will be removed in the note too distant future - the whole of that else branch that you have moved out into an elif and else. My preference would be to keep it as it was since with the outer if/else it, to me, separated more cleanly into two parts the new code that will be kept and the deprecated code that will end up being removed.

jeevesh2002 commented on qiskit/algorithms/amplitude_amplifiers/grover.py:
```diff
@@ -310,35 +310,34 @@ def amplify(self, amplification_problem: AmplificationProblem) -> "GroverResult"
+                top_measurement = np.binary_repr(max_amplitude_idx, num_bits)
+                max_probability = np.abs(max_amplitude) ** 2
+                shots = 1
             else:
```
Undid the changes and added a comment explaining the use of else instead of elif

jeevesh2002 commented on examples/python/initialize.py:
```diff
@@ -14,6 +14,8 @@
 """
 
+
+
```
@jakelishman  formatted the code as per black

ElePT replied:
hi @jeevesh2002, it looks like the formatter is still failing. Could you please run ` tox -e black` and commit again? You should see changes in both files.

ElePT replied:
hi @jeevesh2002, it looks like the formatter is still failing. Could you please run `tox -e black` and commit again? You should see changes in both files.

ElePT reviewed (commented):
hi @jeevesh2002, it looks like the formatter is still failing. Could you please run `tox -e black` and commit again? You should see changes in both files.

jeevesh2002 commented:
Hi @ElePT i ran tox -e black , everything should pass now.

jakelishman reviewed (approved):
Thanks for the quick changes!

For context for others, Jeevesh came to the Qiskit office hours on Monday where we discussed this PR.  The comments I made in the above review were notes from that discussion.

jakelishman commented:
@jeevesh2002: I've tagged this for merge, so at this point a bot takes over and will handle the process of making sure the PR is up-to-date, and will add it to the merge queue.  There's nothing more you should do on this PR, and it'll be released as part of Qiskit Terra 0.23 in two weeks' time.
